packages/block-editor/src/components/iframe/index.js

```diff
@@ -0,0 +1,225 @@
+/**
+ * WordPress dependencies
+ */
+import { useState, useEffect, createPortal, useRef } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import { useResizeObserver } from '@wordpress/compose';
+
+const className = 'editor-styles-wrapper';
+
+/**
+ * Clones stylesheets targetting the editor canvas to the given document. A
+ * stylesheet is considered targetting the editor a canvas if it contains the
+ * `editor-styles-wrapper` or `wp-block` class selectors.
+ *
+ * Ideally, this hook should be removed in the future and styles should be added
+ * explicitly as editor styles.
+ *
+ * @param {Document} doc The document to append cloned stylesheets to.
+ */
+function useStyleSheetsCompat( doc ) {
+	useEffect( () => {
+		// Search the document for stylesheets targetting the editor canvas.
+		Array.from( document.styleSheets ).forEach( ( styleSheet ) => {
+			try {
+				// May fail for external styles.
+				// eslint-disable-next-line no-unused-expressions
+				styleSheet.cssRules;
+			} catch ( e ) {
+				return;
+			}
+
+			const { ownerNode, cssRules } = styleSheet;
+
+			if ( ! cssRules ) {
+				return;
+			}
+
+			const isMatch = Array.from( cssRules ).find(
+				( { selectorText } ) =>
+					selectorText &&
+					( selectorText.includes( `.${ className }` ) ||
+						selectorText.includes( `.wp-block` ) )
```

Comment:
I'm not sure how much of a problem this actually is, but couldn't `selectorText.includes( '.wp-block' )` end up detecting stuff like `.wp-blockz` or `.wp-block-abcde`?

Also, there's an unnecessary usage of backticks here instead of single quotes.

Reply:
It could. It's actually meant to also detect `.wp-block-something` like `.wp-block-paragraph` etc. I'll update the comment.